### PR TITLE
Update ABL scripts to use CORGI

### PR DESCRIPTION
**Pull request by TylerZeroMaster**
part of openstax/ce#2182

This new script effectively performs the same steps as the old transformation script:
- Fetch ABL from https://corgi.ce.openstax.org/api/abl/ 
   - Fetches entries intended for REX where code_version <= `REACT_APP_ARCHIVE`
- Select the newest version of each book
- Select versions that are not in config.books.json

`REACT_APP_ARCHIVE` is read from `src/config.archive-url.json`

The tests show the input and output formats (the output format is the same as before: `{ <uuid>: <commit_sha>, ... }`)

### CODE REVIEW

File: script/validate-abl-import/test.bash [hunk 1]
```diff
@@ -27,8 +27,10 @@ for input_file in "$my_dir"/*.input.json; do
     # Actual test goes here
     # -------------------------
     json=$(cat "$input_file")
-    node "$my_dir/../entry.js" transform-approved-books-data --data "$json" > "$output_file"
+    configured_books="{}"
```
Comment (TomWoodward) [inline script/validate-abl-import/test.bash:30]: shouldn't there be tests that exercise the logic of diffing the books against the existing config?

**Review by TomWoodward — CHANGES_REQUESTED** [review]

File: script/validate-abl-import/test.bash (hunk 1 above)
Comment (TylerZeroMaster) [reply script/validate-abl-import/test.bash]: Thanks for the quick review! Yes, that completely slipped my mind. I added some additional test data for this. The smaller of the new data sets demonstrates what occurs when new versions are present, the other uses some real data to demonstrate that there are no new entries when all ABL versions are already present.

**Review by TomWoodward — APPROVED** [review]

**Comment by TomWoodward** [comment]
i think we should wait to merge this until the current batch of updates is done.

@TylerZeroMaster is the new system being used already? are we good to merge this whenever?

**Comment by TylerZeroMaster** [comment]
> i think we should wait to merge this until the current batch of updates is done.
> 
> @TylerZeroMaster is the new system being used already? are we good to merge this whenever?

@TomWoodward The new system is in production, but it is not being used yet: there is another PR in progress to update Enki/webhosting pipeline to use the new system as well. We can wait to merge that Enki PR alongside this one.

Additionally, if the ABL on Github is updated, the version on CORGI will need to be updated to match. I have a script that will "fast-forward" the CORGI ABL data to match what is present on Github, but I have been running it manually.

Overall, the changes can be merged anytime, but I would appreciate a little bit of warning ahead of time.

File: script/transform-approved-books-data.bash [hunk 2]
```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
+
+set -e
+
+data="$1"
```
[the remark this answers is not in the document]
Comment (TylerZeroMaster) [reply script/transform-approved-books-data.bash]: That is a good point. I originally used inputs like this to separate fetching from transformation and the errors that could occur in each process. I updated the transformation script to read input from 2 files and do some additional checks on the input.

**Comment by TomWoodward** [comment]
@TylerZeroMaster ok, we will shoot to merge this maybe next week after the current batch of updates is done. 

also the "ABL Schema Tests' check is failing with some jq compile error

**Review by RoyEJohnson — APPROVED** [review]

**Comment by TylerZeroMaster** [comment]
> @TylerZeroMaster ok, we will shoot to merge this maybe next week after the current batch of updates is done.
> 
> also the "ABL Schema Tests' check is failing with some jq compile error

Ahh, I think that was because I was using jq version 1.7 which has some changes that must not be backwards compatible. It has been fixed. Let me know when we are ready to merge this change.

**Pull request merged**